src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,52 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MAX_AGE = 130;
+    private static final int MIN_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validate(user);
+        return storageDao.add(user);
+    }
+
+    public void validate(User user) {
+        if (user == null) {
+            throw new InvalidInputDataException("User is null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidInputDataException("Such user is already present in the storage");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidInputDataException("Login can't be null");
+        }
+        if (user.getLogin().trim().equals("")) {
+            throw new InvalidInputDataException("Login can't be []]");
```

Comment:
let's improve message :)
```suggestion
            throw new InvalidInputDataException("Login can't be []");
```